src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -129,27 +137,56 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
   /** Stream of escape keyboard events. */
   private _escapeEventStream = new Subject<void>();
 
+  /** Emits once and only once on the next onStable event. */
+  private readonly _nextStable: Observable<void>;
+
+  /**  The current model value. */
+  private _value: any;
+
   /** View -> model callback called when value changes */
   _onChange: (value: any) => void = () => {};
 
   /** View -> model callback called when autocomplete has been touched */
   _onTouched = () => {};
 
   /* The autocomplete panel to be attached to this trigger. */
-  @Input('matAutocomplete') autocomplete: MatAutocomplete;
+  @Input('matAutocomplete')
+  set _autocomplete(value: MatAutocomplete) {
+    this.autocomplete = value;
+    if (this._displayWithSubscription) {
+      this._displayWithSubscription.unsubscribe();
+    }
+    if (this.autocomplete instanceof MatAutocomplete) {
+      this._displayWithSubscription = this.autocomplete.displayWithChange
+        .subscribe(() => {
+          this.writeValue(this._value);
+        });
+    }
+  }
+  autocomplete: MatAutocomplete;
+  private _displayWithSubscription: Subscription|null = null;
+
+  /** Stream of clicks outside of the autocomplete panel. */
+  private readonly _outsideClickStream: Observable<any>;
 
   constructor(private _element: ElementRef, private _overlay: Overlay,
               private _viewContainerRef: ViewContainerRef,
-              private _zone: NgZone,
+              zone: NgZone,
               private _changeDetectorRef: ChangeDetectorRef,
               @Inject(MAT_AUTOCOMPLETE_SCROLL_STRATEGY) private _scrollStrategy,
               @Optional() private _dir: Directionality,
               @Optional() @Host() private _formField: MatFormField,
-              @Optional() @Inject(DOCUMENT) private _document: any) {}
+              @Optional() @Inject(DOCUMENT) document: Document) {
+    this._nextStable = first.call(zone.onStable);
+    this._outsideClickStream = this._getOutsideClickStream(document);
+  }
 
   ngOnDestroy() {
     this._destroyPanel();
     this._escapeEventStream.complete();
+    if (this._displayWithSubscription) {
```

Comment:
If you initialize the `_displayWithSubscription` to `Subscription.EMPTY`, you can get rid of this null check.

Reply:
Thanks for this suggestion. Code looks much better this way.